kedro/extras/datasets/avro/avro_dataset.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2020 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+# or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""``AVRODataSet`` loads/saves data from/to a AVRO file using an underlying
+filesystem (e.g.: local, S3, GCS). It uses avro library to handle the AVRO file.
+See details about AVRO format: https://avro.apache.org/
+"""
+import json
+from copy import deepcopy
+from pathlib import Path, PurePosixPath
+from typing import Any, Dict, List, Optional
+
+import fsspec  # type: ignore
```

Comment:
What is the `# type: ignore` for here and for the other imports?

Reply:
`mypy` was throwing errors in my venv, it is to skips mypy checks of that lines

Reply:
Okay, thanks for clarifying! We're not actually using `mypy` as a static check for Kedro, so we'd prefer not to have those ignore statements in the code.